presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaMetadata.java

```diff
@@ -284,6 +286,7 @@ private boolean isTablesEnumeratingTable(SchemaTableName schemaTableName)
                         metadata.listViews(session, prefix).stream()))
                 .filter(objectName -> !predicate.isPresent() || predicate.get().test(asFixedValues(objectName)))
                 .map(QualifiedObjectName::asQualifiedTablePrefix)
+                .limit(maxResults)
```

Comment:
How much does it help? Because `metadata.listTables(session, prefix)` and `metadata.listViews(session, prefix)` are already called, you only skip filtering.

Reply:
I saw a very sizable benefit: from my recollection the planning stage of a statement like `select * from information_schema` went from ~30 minutes to a handful of seconds. We have many schemas which are individually listed otherwise. `limit(maxResults)` will short circuit the stream before the `listTables` or `listViews` methods are called for all schemas.

Reply:
Also I would inline `maxResults` to `MAX_PREFIXES_COUNT`